cwltool/process.py

```diff
@@ -107,6 +107,7 @@ def filter(self, record):  # type: (logging.LogRecord) -> bool
     "NetworkAccess",
     "InplaceUpdateRequirement",
     "LoadListingRequirement",
+    "MPIRequirement",
```

Comment:
Should be prefixed like those below
```suggestion
    "http://commonwl.org/cwltool#MPIRequirement",
```

Reply:
Hi Michael - I have tried this and ended up going down a bit of a rabbit hole of confusion about the Schema Salad definitions... I think I've found a bug in SALAD that I'll PR over on that repo (will require a small patch in cwltool also).

Then I noticed that all the requirements with a `http://commonwl.org/cwltool#` prefix also appear with it. Is this what you mean:
```
supportedProcessRequirements = [
    # snip...
    "LoadListingRequirement",
    "MPIRequirement",
    "http://commonwl.org/cwltool#MPIRequirement",
    # snip
    "http://commonwl.org/cwltool#LoadListingRequirement",
    "http://commonwl.org/cwltool#InplaceUpdateRequirement",
]
```
?

Reply:
The prefixes are necessary for requirements that are not (or were not at the time) part of the CWL standard. For now, MPIRequirement must be prefixed only

@tetron can comment on if we should remove the other prefixed requirements that have since moved into a released version of the standard. Shouldn't the upgrader be internally rewriting these, yes?

Reply:
Thanks for the steer - however that takes me back to schema issues... If I make your change, then the tool loading fails in `process.checkRequirements`, because as the tool definition is parsed as:
```
'$namespaces': ordereddict([('cwl', 'http://commonwl.org/cwltool#')]),
 'arguments': ['-c', 'import os; print(os.getpid())'],
 'baseCommand': 'python',
 'class': 'CommandLineTool',
 'cwlVersion': 'v1.2.0-dev3',
 'doc': 'Trivial MPI test that prints the process IDs of each of the parallel\n'
        'processes. Requires Python (but you have cwltool running, right?)\n'
        'and an MPI implementation.\n',
 'http://commonwl.org/cwltool#original_cwlVersion': 'v1.2.0-dev2',
 'id': 'file:///Users/rnash2/work/vestec/cwltool/tests/wf/mpi_simple.cwl',
 'inputs': [],
 'outputs': [ordereddict([('type', 'File'), ('id', 'file:///Users/rnash2/work/vestec/cwltool/tests/wf/mpi_simple.cwl#pids'), ('outputBinding', ordereddict([('glob', 'b32630f55cb7498ca7c8277cb00b49d1856a88dd')]))])],
 'requirements': [ordereddict([('processes', 2), ('class', 'MPIRequirement')])],
 'stdout': 'b32630f55cb7498ca7c8277cb00b49d1856a88dd'}
```

Note that in the .cwl, requirements is:
```
requirements:
  cwl:MPIRequirement:
    processes: 2
```
so the `cwl:` prefix/namespace is getting stripped off somehow. Changing my schema to have `inVocab: false` gives an enormous error during schema validation. At it's root the issue is that the `Names` object passed to the `UnionSchema` initialiser has a default namespace of "http://commonwl", which is correct as an Avro name but not helpful here. This motivates my ticket https://github.com/common-workflow-language/schema_salad/issues/325